slither/detectors/variables/optimize_variable_order.py

```diff
@@ -0,0 +1,219 @@
+from slither.detectors.abstract_detector import AbstractDetector, DetectorClassification
+
+
+class OptimizeVariableOrder(AbstractDetector):
+    """
+    Documentation
+    """
+
+    ARGUMENT = "optimize-var-order"
+    HELP = "Find space optimizations from reordering variables"
+    IMPACT = DetectorClassification.OPTIMIZATION
+    CONFIDENCE = DetectorClassification.HIGH
+
+    WIKI = "https://github.com/trailofbits/slither/wiki/optimal-variable-order"
+    WIKI_TITLE = "Optimal Variable Order"
+    WIKI_DESCRIPTION = "Detect optimal variable order in contract storage layouts and struct definitions to decrease the number of slots used."

```

Comment:
This description seems to indicate that the script will actually find an optimal ordering, but it's not guaranteed. The algorithm is only heuristic, so there may be tighter packings.